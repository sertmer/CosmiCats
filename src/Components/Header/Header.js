import React from 'react';
import './Header.scss';
import alien from '../../images/alien.svg';
import cat from '../../images/cat.svg';

const Header = () => {
  return (
    <header>
      <h1>The Center for CosmiCat Research</h1>
      <div>
        <img className="icon" src={alien} alt='small green alien icon' />
        <img className="icon" src={cat} alt='small green cat icon' />
      </div>
    </header>
  )
}

export default Header;