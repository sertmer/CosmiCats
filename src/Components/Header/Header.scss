@import '../../_scss/variables';
@import '../../_scss/mixins';

header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.icon {
  height: 5vh;
}

h1 {
  color: white;
  font-family: 'Audiowide', cursive;;
}