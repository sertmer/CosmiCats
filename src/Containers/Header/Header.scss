@import '../../_scss/variables';
@import '../../_scss/mixins';

header {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.icon {
  height: 5vh;
}

.cccr-header {
  color: white;
  font-family: $accent-font;
  margin: 0 5% 0 0;
  width: 90%;
}

.triple-equals {
  font-size: x-large;
  font-family: $accent-font;
  color: white;
}

.icon-div {
  display: flex;
  flex-direction: row;
  width: 5%;
}

.user-name {
  display: flex;
  flex-direction: row;
  width: 100px;
  justify-content: space-between;
}

.header-h2 {
  margin: 0;
}