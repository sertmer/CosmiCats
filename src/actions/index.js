export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const setNasaImages = images => ({
  type: 'SET_NASA_IMAGES',
  images
})

export const setCatImages = images => ({
  type: 'SET_CAT_IMAGES',
  images
})

export const addToEvidence = evidence => ({
  type: 'ADD_EVIDENCE',
  evidence
})