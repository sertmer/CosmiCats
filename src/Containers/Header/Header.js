import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { setUser } from '../../actions';
import './Header.scss';
import alien from '../../images/alien.svg';
import cat from '../../images/cat.svg';

export const Header = ({ user, setUser, evidence }) => {
  return (
    <header>
      <h1 className='cccr-header'>The Center for CosmiCat Research</h1>
      {user &&
        <div className='user-name'>
          <h2 className='header-h2'>Welcome Dr. {user}</h2>
          <Link to='/'>
            <button className='signout-btn' onClick={() => setUser('')}>Sign Out</button>
          </Link>
        </div>
      }
      {evidence.length > 0 && 
        <Link to='/evidence'>
          <button className='evidence-btn' type='button'>Evidence</button>
        </Link>
      }
      <div className='icon-div'>
        <img className="icon" src={alien} alt='small green alien icon' />
        <img className="icon" src={cat} alt='small green cat icon' />
      </div>
    </header>
  )
}

export const mapStateToProps = state => ({
  user: state.user,
  evidence: state.evidence
})

export const mapDispatchToProps = dispatch => ({
  setUser: user => dispatch(setUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);