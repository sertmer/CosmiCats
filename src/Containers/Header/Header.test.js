import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { setUser } from '../../actions'

describe('Header', () => {

  let wrapper
  let mockSetUser
  let mockEvidence

  beforeEach(() => {
    mockEvidence = [{id:'ab1', img: 'someURL.com', dateCreated: '12-12-12'}]
    mockSetUser = jest.fn()
    wrapper = shallow(<Header user={'Davey Havok'} setUser={mockSetUser} evidence={mockEvidence}/>)
  })

  describe('Header unit tests', () => {

    it('should match the snapshot', () => {
      wrapper = shallow(<Header evidence={mockEvidence}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot when there is a user in props', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call setUser on signout click', () => {
     
      wrapper.find('.signout-btn').simulate('click');

      expect(mockSetUser).toHaveBeenCalledWith('')
    })
  })

  describe('mapStateToProps', () => {
    it('should return the users name', () => {
      const mockState = {
        user: 'Davey Havok',
        extra: 'extra'
      }
      const expected = { user: 'Davey Havok'}
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
  
  describe('mapDispatchToProps', () => {
    let mockDispatch, mockActionToDispatch

    beforeEach(() => {
      mockDispatch = jest.fn()
    })

    it('should call mapDispatchToProps with an empty string user', () => {
      mockActionToDispatch = setUser('')
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.setUser('')
      expect(mockDispatch).toHaveBeenCalledWith(mockActionToDispatch)
    })
  })
})