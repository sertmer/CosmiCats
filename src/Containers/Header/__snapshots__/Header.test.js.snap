// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header unit tests should match the snapshot 1`] = `
<header>
  <h1
    className="cccr-header"
  >
    The Center for CosmiCat Research
  </h1>
  <div
    className="icon-div"
  >
    <img
      alt="small green alien icon"
      className="icon"
      src="alien.svg"
    />
    <p
      className="triple-equals"
    >
      ===
    </p>
    <img
      alt="small green cat icon"
      className="icon"
      src="cat.svg"
    />
  </div>
</header>
`;

exports[`Header Header unit tests should match the snapshot when there is a user in props 1`] = `
<header>
  <h1
    className="cccr-header"
  >
    The Center for CosmiCat Research
  </h1>
  <div>
    <h2>
      Welcome Dr. 
      Davey Havok
    </h2>
    <Link
      to="/"
    >
      <button
        onClick={[Function]}
      >
        Sign Out
      </button>
    </Link>
  </div>
  <div
    className="icon-div"
  >
    <img
      alt="small green alien icon"
      className="icon"
      src="alien.svg"
    />
    <p
      className="triple-equals"
    >
      ===
    </p>
    <img
      alt="small green cat icon"
      className="icon"
      src="cat.svg"
    />
  </div>
</header>
`;
